src/tree/definition/type.rs

```diff
@@ -42,6 +40,7 @@ pub enum TypeDefinition {
     Iterable(usize, TypeTemplateGroupDefinition),
     Class(usize, TypeTemplateGroupDefinition),
     Interface(usize, TypeTemplateGroupDefinition),
+    Literal(usize, Literal),
```

Comment:
```suggestion
    Literal(Literal),
```

the literal contains it's own position, so there's no need to keep track of this.